brushfire-core/src/main/scala/com/stripe/brushfire/TreeTraversal.scala

```diff
@@ -15,12 +16,14 @@ import com.twitter.algebird._
  * features). A tree traversal chooses which paths to go down (which may be all
  * of them) and the order in which they are traversed.
  */
-trait TreeTraversal[K, V, T, A] {
+trait TreeTraversal[Tree, K, V, T, A] {
```

Comment:
Maybe rename `Tree` to `TreeRep`? Then we can either add the alias back without much confusion, or just rename `AnnotatedTree` to `Tree`. WDYT?